### DISCUSSION TITLE
Where to store role specific data?

### DISCUSSION THREAD

**Discussion by bilogic** [spatie/laravel-permission#1969] [Q&A]
Suppose I have users with very diverse data based on their roles, is there a suggested way on where to store these role specific data so that it is still easily accessible through the models?

**Comment by angeljqv** [comment]
There isn't any "role specific data", but you can customize everything, there is no suggestions for that

**Comment by erikn69** [comment]
Maybe inside `model_has_roles` and adding `withPivot` on relation, but for that you would need to override the methods(`roles()`,`assignRole(...$roles)`) on user Model, example overrride: https://github.com/spatie/laravel-permission/discussions/1895#discussioncomment-1523163
```php
public function roles(): BelongsToMany
{
    return $this->rolesTrait()->withPivot(['YOUR','CUSTOM','DATA']);
}
```
----
https://github.com/spatie/laravel-permission/blob/79f7dbf2b6b21d81209774f5726a0e9f4222dd92/src/Traits/HasRoles.php#L122-L123
```php
$array[$role->getKey()] = ['YOUR'=>'CUSTOM', 'DATA'=>''] + (PermissionRegistrar::$teams && ! is_a($this, Permission::class) ?
                    [PermissionRegistrar::$teamsKey => getPermissionsTeamId()] : []);
```
----
Or better add your own sync method
